tests/driver/runner_test.py

```diff
@@ -15,24 +16,36 @@ def runner_function(es, params):
 
         runner.register_runner(operation_type="unit_test", runner=runner_function)
         returned_runner = runner.runner_for("unit_test")
-        self.assertIsInstance(returned_runner, runner.DelegatingRunner)
+        self.assertIsInstance(returned_runner, runner.SingleClusterDelegatingRunner)
         self.assertEqual("user-defined runner for [runner_function]", repr(returned_runner))
 
-    def test_runner_class_with_context_manager_should_be_registered_as_is(self):
-        class UnitTestRunner:
-            def __enter__(self):
+    def test_runner_class_with_context_manager_should_be_wrapped_with_context_manager_enabled(self):
+        class UnitTestContextManagerRunner:
+            def __call__(self, *args):
                 return self
 
-            def __call__(self, *args):
-                pass
+            def __enter__(self):
+                self.fp = io.StringIO("many\nlines\nin\na\nfile")
+                return self.fp
```

Comment:
I think you should still return `self` here. However, you need to forward method calls to the delegate (i.e. `self.fp`).

Reply:
Any particular reason for this? My understanding, and based on what I've read, this is ok and pretty standard and lets context manager definitions use the simpler `with ... as myobject` and then just reference `myobject.method()` which is for example what we do with Python's native `open()` rather than something like `as myobject` and later on `myobject.fp.read()`.

For example this:

```
import io        
                       
class Foo:
    def __enter__(self):
        self.fp = io.StringIO("many\nlines\nin\na\nfile")
        return self
    def __exit__(self, a, b, c):
        self.fp.close()
        return False
```

needs to be used as:

```
with Foo() as foo:
    print(foo.fp.read())
```

rather than:

```
with Foo() as foo:
    print(foo.read())
```

Reply:
I thought that the way it is implemented now the delegate actually escapes from the wrapper.